### PR TITLE
[DNS] Zone conversions guides

### PR DESCRIPTION
**Pull request by RebeccaTamachiro**
### Summary

PCX-8015: Document zone setup conversions

### Documentation checklist

- [x] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.
- [x] Files which have changed name or location have been allocated [redirects](https://developers.cloudflare.com/pages/configuration/redirects/#per-file).

### CODE REVIEW

File: src/content/docs/dns/zone-setups/conversions/convert-full-to-secondary.mdx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    </TabItem> </Tabs>
+
+3. Create a peer server.
+
+    <Render file="create-peer-server" />
+
+## 3. Convert the zone and initiate zone transfers
+
+1. Use the [Edit Zone endpoint](/api/operations/zones-0-patch) with `type` set to `secondary`, to convert the zone type. The existing records will remain in place.
```
Comment (Oxyjun) [inline src/content/docs/dns/zone-setups/conversions/convert-full-to-secondary.mdx]:
```suggestion
1. Use the [Edit Zone endpoint](/api/operations/zones-0-patch) with `type` set to `secondary` to convert the zone type. The existing records will remain in place.
```

File: src/content/docs/dns/zone-setups/conversions/convert-full-to-secondary.mdx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+2. Go to **DNS** > **Settings** > **DNS Zone Transfers** and select **Manage linked peers**.
+3. Link the peer server you created in the previous steps and select **Save**.
+4. On  **DNS** > **Settings**, select **Initiate zone transfer**.
+5. Confirm the DNS records are transferring as expected.
+6. Go to [**DNS** > **Records**](https://dash.cloudflare.com/?to=/:account/:zone/dns/records) and take note of your new **Cloudflare Nameservers**.
```
Comment (Oxyjun) [inline src/content/docs/dns/zone-setups/conversions/convert-full-to-secondary.mdx:55]:
```suggestion
6. Go to [**DNS** > **Records**](https://dash.cloudflare.com/?to=/:account/:zone/dns/records) and take note of your new **Cloudflare nameservers**.
```
I could be wrong on this, but in the same way we say `Cloudflare dashboard` (lowercase "d"), I thought maybe this should be `Cloudflare nameservers`, with a lowercase n.

File: src/content/docs/dns/zone-setups/conversions/convert-secondary-to-full.mdx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+Follow the steps below to achieve this conversion.
+
+## 1. Stop transferring the zone
+
+<Render file="secondary-conversion-unlink" />
+
+## 2. Prepare for the conversion
+
+1. Plan for [DNSSEC settings](/dns/zone-setups/zone-transfers/cloudflare-as-secondary/dnssec-for-secondary/). If you were previously using [Pre-signed DNSSEC](/dns/zone-setups/zone-transfers/cloudflare-as-secondary/dnssec-for-secondary/#set-up-pre-signed-dnssec), consider disabling DNSSEC before starting the conversion.
```
Comment (Oxyjun) [inline src/content/docs/dns/zone-setups/conversions/convert-secondary-to-full.mdx:24]:
```suggestion
1. Plan for [DNSSEC settings](/dns/zone-setups/zone-transfers/cloudflare-as-secondary/dnssec-for-secondary/). If you were previously using [presigned DNSSEC](/dns/zone-setups/zone-transfers/cloudflare-as-secondary/dnssec-for-secondary/#set-up-pre-signed-dnssec), consider disabling DNSSEC before starting the conversion.
```

I've seen both `presigned` and `pre-signed` being used in the docs. Happy to be overruled here.

File: src/content/docs/dns/zone-setups/conversions/convert-secondary-to-full.mdx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+2. Make sure the [proxy statuses](/dns/manage-dns-records/reference/proxied-dns-records/) of your DNS records are consistently set:
+
+    - If you have [Secondary DNS override](/dns/zone-setups/zone-transfers/cloudflare-as-secondary/proxy-traffic/), confirm each record has the appropriate setting (**Proxied** or **DNS only**).
```
Comment (Oxyjun) [inline src/content/docs/dns/zone-setups/conversions/convert-secondary-to-full.mdx:27]:
```suggestion
    - If you have [secondary DNS override](/dns/zone-setups/zone-transfers/cloudflare-as-secondary/proxy-traffic/), confirm each record has the appropriate setting (**Proxied** or **DNS only**).
```

File: src/content/docs/dns/zone-setups/conversions/convert-secondary-to-full.mdx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    - If [Secondary DNS override](/dns/zone-setups/zone-transfers/cloudflare-as-secondary/proxy-traffic/) is disabled, make sure all of your DNS records are listed as **DNS only**.
```
Comment (Oxyjun) [inline src/content/docs/dns/zone-setups/conversions/convert-secondary-to-full.mdx:28]:
```suggestion
    - If [secondary DNS override](/dns/zone-setups/zone-transfers/cloudflare-as-secondary/proxy-traffic/) is disabled, make sure all of your DNS records are listed as **DNS only**.
```

File: src/content/docs/dns/zone-setups/conversions/convert-secondary-to-full.mdx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+3. (Optional) For consistency, use the [Update DNS Settings](/api/operations/dns-settings-for-a-zone-update-dns-settings) endpoint to specify SOA record fields according to your needs. Once Cloudflare automatically generates an SOA record for your zone on full setup, the field overrides will be considered.
+
+## 3. Convert your zone
+
+1. Use the [Edit Zone endpoint](/api/operations/zones-0-patch) with `type` set to `full`, to convert the zone type. Existing DNS records will not be affected.
```
Comment (Oxyjun) [inline src/content/docs/dns/zone-setups/conversions/convert-secondary-to-full.mdx]:
```suggestion
1. Use the [Edit Zone endpoint](/api/operations/zones-0-patch) with `type` set to `full` to convert the zone type. Existing DNS records will not be affected.
```

File: src/content/docs/dns/zone-setups/conversions/convert-secondary-to-full.mdx [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+2. Go to [**DNS** > **Records**](https://dash.cloudflare.com/?to=/:account/:zone/dns/records) and take note of your new **Cloudflare Nameservers**.
```
Comment (Oxyjun) [inline src/content/docs/dns/zone-setups/conversions/convert-secondary-to-full.mdx:35]:
```suggestion
2. Go to [**DNS** > **Records**](https://dash.cloudflare.com/?to=/:account/:zone/dns/records) and take note of your new **Cloudflare nameservers**.
```

File: src/content/docs/dns/zone-setups/conversions/convert-secondary-to-partial.mdx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+Follow the steps below to achieve this conversion.
+
+## 1. Stop transferring the zone
+
+<Render file="secondary-conversion-unlink" />
+
+## 2. Configure your authoritative DNS provider
+
+1. (Optional) If you are also migrating to a new authoritative DNS provider, export a zone file from the previous provider and import into the new one.
```
Comment (Oxyjun) [inline src/content/docs/dns/zone-setups/conversions/convert-secondary-to-partial.mdx]:
```suggestion
1. (Optional) If you are also migrating to a new authoritative DNS provider, export a zone file from the previous provider and import it into the new one.
```

File: src/content/docs/dns/zone-setups/conversions/convert-secondary-to-partial.mdx [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    www.example.com CNAME www.example.com.cdn.cloudflare.net
+    ```
+    </Details>
+
+3. At your authoritative DNS provider, remove any previously existing `A`, `AAAA`, or `CNAME` records referencing the hostnames you want to proxy through Cloudflare. For these hostnames, leave only the records pointing to `{your-hostname}.cdn.cloudflare.net`.
+
+## 3. Convert your Cloudflare zone
+
+1. Back at your Cloudflare zone, confirm that you have all the `A`, `AAAA`, or `CNAME` [DNS records](/dns/manage-dns-records/how-to/create-dns-records/) needed for the hostnames you pointed to `{your-hostname}.cdn.cloudflare.net` in the previous step. You can also delete any DNS records that have a different type, as they will no longer resolve once you convert your zone to a partial setup.
+2. Use the [Edit Zone endpoint](/api/operations/zones-0-patch) with `type` set to `partial`, to convert the zone type. Existing DNS records will not be affected.
```
Comment (Oxyjun) [inline src/content/docs/dns/zone-setups/conversions/convert-secondary-to-partial.mdx]:
```suggestion
2. Use the [Edit Zone endpoint](/api/operations/zones-0-patch) with `type` set to `partial` to convert the zone type. Existing DNS records will not be affected.
```

**Review by Oxyjun — APPROVED** [review]

File: src/content/docs/dns/zone-setups/conversions/convert-full-to-secondary.mdx (hunk 2 above)
Comment (RebeccaTamachiro) [reply src/content/docs/dns/zone-setups/conversions/convert-full-to-secondary.mdx:55]: I agree with this reasoning and have used `nameservers` elsewhere, but have kept it like this in these guides to mirror the card name in the dashboard.

File: src/content/docs/dns/zone-setups/conversions/convert-secondary-to-full.mdx (hunk 4 above)
Comment (RebeccaTamachiro) [reply src/content/docs/dns/zone-setups/conversions/convert-secondary-to-full.mdx:27]: I think we've been consistently using the capitalized form, as this is considered the name of a Cloudflare feature.

**Pull request merged**